jpsreport/general/ArgumentParser.cpp

```diff
@@ -592,6 +585,138 @@ bool ArgumentParser::ParseIniFile(const string & inifile)
+            polygon_2d subroom;
+
+            string subroom_id = xmltoa(xSubRoom->Attribute("id"), "-1");
+
+            //looking for polygons (walls)
+            for(TiXmlElement * xPolyVertices = xSubRoom->FirstChildElement("polygon");
+                xPolyVertices;
+                xPolyVertices = xPolyVertices->NextSiblingElement("polygon")) {
+                for(TiXmlElement * xVertex = xPolyVertices->FirstChildElement("vertex");
+                    xVertex && xVertex != xPolyVertices->LastChild("vertex");
+                    xVertex = xVertex->NextSiblingElement("vertex")) {
+                    double x1 = xmltof(xVertex->Attribute("px"));
+                    double y1 = xmltof(xVertex->Attribute("py"));
+                    double x2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("px"));
+                    double y2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("py"));
+                    append(subroom, make<point_2d>(x1 * M2CM, y1 * M2CM));
+                    append(subroom, make<point_2d>(x2 * M2CM, y2 * M2CM));
+                }
+            }
+            correct(subroom);
+
+            std::vector<polygon_2d> obstacles;
+            //looking for obstacles
+            for(TiXmlElement * xObstacle = xSubRoom->FirstChildElement("obstacle"); xObstacle;
+                xObstacle                = xObstacle->NextSiblingElement("obstacle")) {
+                //looking for polygons (walls)
+                polygon_2d obstacle;
+
+                for(TiXmlElement * xPolyVertices = xObstacle->FirstChildElement("polygon");
+                    xPolyVertices;
+                    xPolyVertices = xPolyVertices->NextSiblingElement("polygon")) {
+                    for(TiXmlElement * xVertex = xPolyVertices->FirstChildElement("vertex");
+                        xVertex && xVertex != xPolyVertices->LastChild("vertex");
+                        xVertex = xVertex->NextSiblingElement("vertex")) {
+                        double x1 = xmltof(xVertex->Attribute("px"));
+                        double y1 = xmltof(xVertex->Attribute("py"));
+                        double x2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("px"));
+                        double y2 = xmltof(xVertex->NextSiblingElement("vertex")->Attribute("py"));
+
+                        append(obstacle, make<point_2d>(x1 * M2CM, y1 * M2CM));
+                        append(obstacle, make<point_2d>(x2 * M2CM, y2 * M2CM));
+                    }
+                }
+                correct(obstacle);
+            }
+
+            int k = 1;
+            for(auto && obstacle : obstacles) {
+                subroom.inners().resize(k++);
+                subroom.inners().back();
+                model::ring<point_2d> & inner = subroom.inners().back();
+                for(auto && tmp_point : obstacle.outer()) {
+                    append(inner, make<point_2d>(tmp_point.x() * M2CM, tmp_point.y() * M2CM));
+                }
```

Comment:
I'm not sure what's happening here and how this is affecting the calculation methods. Could you add a comment?

Reply:
Added comment